src/mux-analytics.brs

```diff
@@ -535,6 +584,122 @@ function muxAnalytics() as Object
     end if
   end sub
 
+  prototype.videoContentIndexChangeHandler = sub(contentIndex as integer)
+    if contentIndex > 0
+      m._addEventToQueue(m._createEvent("ended"))
```

Comment:
we don't actually want the `ended` event unless playback has made it to the end of the video. Is there a way for us to check that here, or would we get a completion event elsewhere to emit `ended`?

Reply:
The above will be invoked when the current content ends or is skipped and the new content starts playing. We don't have any other place to check when the current playback ends.

Reply:
Right, so what I was thinking here was if we could check if we reached the end of that first content, rather than if it was skipped. If skipped, we wouldn't want to emit the ended event.

Something like: 

```
if contentIndex > 0
  if player_playhead_time >= duration_of_content
    m._addEventToQueue(m._createEvent("ended")
  end if
  
  m._endView(true)
  m._startView(true)
  ... ect
```

Reply:
The playlist's index change handler is invoked when the video player starts the new content not when the current content ends. So, the player_playhead_time will always be 0.